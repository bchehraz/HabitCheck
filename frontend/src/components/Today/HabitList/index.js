import Emoji from "./Emoji"
import HabitList from "./HabitList"
import Tag from "./Tag"
import AddHabitButton from "./AddHabitButton"
import ListItemController from "./ListItemController"
import ListItem from "./ListItem"
import ListItemMobile from "./ListItemMobile"

export {
  Emoji,
  HabitList,
  Tag,
  AddHabitButton,
  ListItemController,
  ListItem,
  ListItemMobile,
}
