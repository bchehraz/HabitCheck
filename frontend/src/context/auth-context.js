import React from "react"

/* eslint-disable */

const AuthContext = React.createContext({
  token: null,
  userId: null,
  email: null,
  data: {
    habits: {
      lastUpdate: null,
      checked: [],
      unchecked: [],
      map: [],
    },
  },
  preferences: {
    darkMode: false,
    xEffectView: false,
  },
  login: (token, userId, email, data, preferences) => {},
  logout: () => {},
  addHabit: habit => {},
  checkHabit: (habitIndex, callback) => {},
  uncheckHabit: (habitIndex, callback) => {},
  toggleDarkMode: () => {},
  toggleXEffectView: () => {},
  changeHabitTitle: (title, newTitle) => {},
})

export const AuthProvider = AuthContext.Provider
export const AuthConsumer = AuthContext.Consumer
export default AuthContext
