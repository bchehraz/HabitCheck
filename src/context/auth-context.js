import React from "react"

const AuthContext = React.createContext({
  token: null,
  userId: null,
  email: null,
  tokenExpiration: null,
  data: {
    habits: {
      lastUpdate: null,
      checked: [],
      unchecked: [],
      map: [],
    },
  },
  preferences: {
    darkMode: false,
    xEffectView: false,
    selected: 0,
  },
  // eslint-disable-next-line
  login: (token, userId, tokenExpiration, email, data, preferences) => {},
  logout: () => {},
  newHabit: habit => {},
  checkHabit: (habitIndex, callback) => {},
  uncheckHabit: (habitIndex, callback) => {},
  toggleDarkMode: () => {},
  toggleXEffectView: () => {},
})

export const AuthProvider = AuthContext.Provider
export const AuthConsumer = AuthContext.Consumer
export default AuthContext
