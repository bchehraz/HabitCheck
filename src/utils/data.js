export const data = {
  habits: {
    unchecked: [
      {
        title: "Work on HabitCheck",
        startDate: "12/02/2019",
        bestStreak: 0,
        progress: [],
      },
      {
        title: "TEST_DATA_1",
        startDate: "12/02/2019",
        bestStreak: 0,
        progress: [{ date: "2/6/2020", streak: 24 }],
      },
      {
        title: "Quit Smoking",
        startDate: "10/25/2019",
        bestStreak: 40,
        bestStreakDate: "12/03/2019",
        progress: [
          {
            date: "10/25/2019",
            streak: 40,
          },
          //{ date: "12/4/2019", streak: -1 }
        ],
      },
      {
        title: "Meditate",
        startDate: "11/28/2019",
        bestStreak: 1,
        bestStreakDate: "11/28/2019",
        progress: [
          { date: "11/28/2019", streak: 1 },
          { date: "11/29/2019", streak: -3 },
        ],
      },
      {
        title: "Sleep on time",
        startDate: "11/29/2019",
        bestStreak: 1,
        bestStreakDate: "11/30/2019",
        progress: [
          { date: "11/30/2019", streak: 1 },
          { date: "12/1/2019", streak: -1 },
        ],
      },
    ],
    checked: [
      {
        title: "90 min Yoga",
        startDate: "12/1/2019",
        bestStreak: 1,
        bestStreakDate: "12/1/2019",
        progress: [
          {
            date: "12/1/2019",
            streak: 1,
          },
        ],
      },
      {
        title: "Push Ups",
        startDate: "12/1/2019",
        bestStreak: 1,
        bestStreakDate: "12/1/2019",
        progress: [
          {
            date: "12/1/2019",
            streak: 1,
          },
        ],
      },
      {
        title: "30 min walk",
        startDate: "12/1/2019",
        bestStreak: 1,
        bestStreakDate: "12/1/2019",
        progress: [
          {
            date: "12/1/2019",
            streak: 1,
          },
        ],
      },
      {
        title: "Keto Diet",
        startDate: "12/1/2019",
        bestStreak: 3,
        bestStreakDate: "12/3/2019",
        progress: [
          {
            date: "12/1/2019",
            streak: 3,
          },
        ],
      },
    ],
  },
}

export const data_1 = {
  habits: {
    checked: [],
    unchecked: [],
  },
}

let date = new Date()
date.setDate(date.getDate() - 45)

export const data_2 = {
  habits: {
    unchecked: [
      {
        title: "Work on HabitCheck",
        startDate: "12/02/2019",
        bestStreak: 0,
        progress: [],
      },
      {
        title: "TEST_DATA_1",
        startDate: "12/02/2019",
        bestStreak: 0,
        progress: [{ date: "2/6/2020", streak: 24 }],
      },
      {
        title: "Quit Smoking",
        startDate: "10/25/2019",
        bestStreak: 40,
        bestStreakDate: "12/03/2019",
        progress: [
          {
            date: "10/25/2019",
            streak: 40,
          },
          //{ date: "12/4/2019", streak: -1 }
        ],
      },
      {
        title: "Meditate",
        startDate: "11/28/2019",
        bestStreak: 1,
        bestStreakDate: "11/28/2019",
        progress: [
          { date: "11/28/2019", streak: 1 },
          { date: "11/29/2019", streak: -3 },
        ],
      },
      {
        title: "Sleep on time",
        startDate: "11/29/2019",
        bestStreak: 1,
        bestStreakDate: "11/30/2019",
        progress: [
          { date: "11/30/2019", streak: 1 },
          { date: "12/1/2019", streak: -1 },
        ],
      },
    ],
    checked: [
      {
        title: "90 min Yoga",
        startDate: "12/1/2019",
        bestStreak: 1,
        bestStreakDate: "12/1/2019",
        progress: [
          {
            date: "12/1/2019",
            streak: 1,
          },
        ],
      },
      {
        title: "Push Ups",
        startDate: "12/1/2019",
        bestStreak: 1,
        bestStreakDate: "12/1/2019",
        progress: [
          {
            date: "12/1/2019",
            streak: 1,
          },
        ],
      },
      {
        title: "30 min walk",
        startDate: "12/1/2019",
        bestStreak: 1,
        bestStreakDate: "12/1/2019",
        progress: [
          {
            date: "12/1/2019",
            streak: 1,
          },
        ],
      },
      {
        title: "Keto Diet",
        startDate: "12/1/2019",
        bestStreak: 3,
        bestStreakDate: "12/3/2019",
        progress: [
          {
            date: "12/1/2019",
            streak: 3,
          },
        ],
      },
    ],
  },
}

/*

let today = new Date();
lastDate = new Date("11/3/2019");
diffTime = Math.abs(today - lastDate);

lastDate2 = new Date("11/4/2019");
diffTime2 = Math.abs(today - lastDate2)
diffTime
diffTime2

let diffDays = Math.ceil(diffTime / (1000 * 60 * 60 * 24))
let diffDays2 = Math.ceil(diffTime2 / (1000 * 60 * 60 * 24)) 

const streak = 40;

let newStreak = diffDays - streak;
let newStreak2 = diffDays2 - streak;

let missedDay = new Date(today)
let missedDay2 = new Date(today)
missedDay.setDate(missedDay.getDate() - newStreak)
missedDay2.setDate(missedDay2.getDate() - newStreak2)


//missedDay = December 12
//missedDay2 = December 13

//-----------> diffDays, streak, diffDays-streak
//11/3/2019 -> 110       40      70
//11/4/2019 -> 108       40      68


Math.ceil(Math.abs(new Date() - date1) / (1000 * 60 * 60 * 24))
*/
