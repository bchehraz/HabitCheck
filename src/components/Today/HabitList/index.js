import Emoji from "./Emoji"
import HabitList from "./HabitList"
import HabitListItem from "./HabitListItem"
import Tag from "./Tag"
import AddHabitButton from "./AddHabitButton"

export { Emoji, HabitList, HabitListItem, Tag, AddHabitButton }
