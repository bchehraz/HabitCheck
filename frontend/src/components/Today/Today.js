import React, { useState } from "react"
import PropTypes from "prop-types"
import styled from "styled-components"

import { AppLayout } from "../Layout"
import { HabitList } from "./HabitList"
import HabitStats from "../HabitStats/HabitStats"
import Backdrop from "../Backdrop"
import { AuthConsumer } from "../../context/auth-context"

const Modal = styled.div`
  position: fixed;
  top: 0;
  left: 0;
  opacity: ${({ show }) => (show ? 1 : 0)};
  visibility: ${({ show }) => (show ? "visible" : "hidden")};
  background-color: rgba(0, 0, 0, 0);
  margin: 0 auto;

  transition: all 300ms ease-out;
  z-index: 2000;

  width: 100vw;
  height: auto;
  padding: 0;
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 0 0 20px;

  top: 50%;
  bottom: 50%;

  @media only screen and (min-width: 600px) {
    margin-left: -200px;
    max-width: 400px;
    left: 50%;
  }
`

const Today = ({ path }) => {
  const [showStats, setShowStats] = useState(false)
  const [selectedTitle, setSelectedTitle] = useState("")

  const selectHabit = title => {
    setShowStats(true)
    setSelectedTitle(title)
  }

  return (
    <AuthConsumer>
      {context => (
        <AppLayout path={path}>
          <HabitList
            habits={context.data.habits}
            checkHabit={title => context.checkHabit(title)}
            uncheckHabit={title => context.uncheckHabit(title)}
            handleAddHabit={habitTitle => {
              context.newHabit(habitTitle)
            }}
            onViewStats={selectHabit}
          />
          <Modal show={showStats}>
            {showStats && <HabitStats title={selectedTitle} show={showStats} />}
          </Modal>
          <Backdrop
            enabled={showStats}
            onClick={() => setShowStats(false)}
            zIndex={1999}
          />
        </AppLayout>
      )}
    </AuthConsumer>
  )
}

Today.propTypes = {
  path: PropTypes.string.isRequired,
}

export default Today
