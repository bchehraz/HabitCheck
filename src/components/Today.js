import React, { useState } from "react"
import PropTypes from "prop-types"
import styled from "styled-components"

import AppLayout from "./AppLayout"
import HabitList from "./HabitList"
import HabitStats from "./HabitStats"
import { AuthConsumer } from "../context/auth-context"

const HiddenPage = styled.div`
  position: relative;
  top: 0;
  left: 0;
  opacity: ${({ show }) => +show};
  background-color: white;
  width: 100vw;
  height: 100vh;
  max-width: 400px;
  margin: 0 auto;
`

const Today = ({ path }) => {
  const [showStats, setShowStats] = useState(false)
  const [selectedTitle, setSelectedTitle] = useState("")

  const selectHabit = title => {
    setShowStats(true)
    setSelectedTitle(title)
  }

  return (
    <AuthConsumer>
      {context => (
        <AppLayout path={path}>
          {showStats && (
            <HiddenPage show={showStats}>
              <button onClick={() => setShowStats(false)} className="backBtn">
                Back
              </button>
              <HabitStats title={selectedTitle} />
            </HiddenPage>
          )}
          {!showStats && (
            <HabitList
              habits={context.data.habits}
              checkHabit={index => context.checkHabit(index)}
              uncheckHabit={index => context.uncheckHabit(index)}
              handleAddHabit={habitTitle => {
                context.newHabit(habitTitle)
              }}
              onViewStats={selectHabit}
            />
          )}
        </AppLayout>
      )}
    </AuthConsumer>
  )
}

Today.propTypes = {
  path: PropTypes.string.isRequired,
}

export default Today
