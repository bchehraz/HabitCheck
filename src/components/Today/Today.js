import React, { useState } from "react"
import PropTypes from "prop-types"
import styled from "styled-components"

import { AppLayout } from "../Layout"
import { HabitList } from "./HabitList"
import HabitStats from "../HabitStats/HabitStats"
import Backdrop from "../Backdrop"
import { AuthConsumer } from "../../context/auth-context"

const Modal = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  opacity: ${({ show }) => +show};
  background-color: white;
  margin: 0 auto;
  z-index: ${props => (props.show ? 56 : -1)};
  transition: all 300ms ease-in-out;
  border-radius: 17px;
  box-shadow: 3px 3px 7px rgba(0, 0, 0, 0.29);

  width: 100vw;
  padding: 0;
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 20px 0;

  @media only screen and (min-width: 600px) {
    margin-left: -200px;
    max-width: 400px;
    left: 50%;
  }
`

const Today = ({ path }) => {
  const [showStats, setShowStats] = useState(false)
  const [selectedTitle, setSelectedTitle] = useState("")

  const selectHabit = title => {
    setShowStats(true)
    setSelectedTitle(title)
  }

  return (
    <AuthConsumer>
      {context => (
        <AppLayout path={path}>
          <HabitList
            habits={context.data.habits}
            checkHabit={title => context.checkHabit(title)}
            uncheckHabit={title => context.uncheckHabit(title)}
            handleAddHabit={habitTitle => {
              context.newHabit(habitTitle)
            }}
            onViewStats={selectHabit}
          />
          {showStats && (
            <Modal show={showStats}>
              <HabitStats
                viewPref2={context.preferences.xEffectView}
                title={selectedTitle}
              />
            </Modal>
          )}
          <Backdrop
            enabled={showStats}
            onClick={() => setShowStats(false)}
            zIndex={50}
          />
        </AppLayout>
      )}
    </AuthConsumer>
  )
}

Today.propTypes = {
  path: PropTypes.string.isRequired,
}

export default Today
