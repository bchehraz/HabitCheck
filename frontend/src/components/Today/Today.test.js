import React from "react"
import { render, fireEvent } from "@testing-library/react"
import renderer from "react-test-renderer"
import { advanceTo, clear } from "jest-date-mock"

import Today from "./Today.js"
import { AuthProvider } from "../../context/auth-context"
import { data } from "../../utils/static/data"

describe("Today.js Snapshot Test", () => {
  // Mock date prior to running snapshot test since the date is visible in the component
  beforeEach(() => advanceTo(new Date("7/4/2020")))

  test("Today: Snapshot test", () => {
    const tree = renderer
      .create(<Today path="/app/" handleAddHabit={() => {}} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
    clear()
  })
})

test("Today: Render with empty habit data, fresh context", () => {
  const { queryAllByTestId } = render(
    <AuthProvider
      value={{
        data: {
          habits: {
            lastUpdate: null,
            checked: [],
            unchecked: [],
            map: [],
          },
        },
      }}
    >
      <Today path="/app/" handleAddHabit={() => {}} />
    </AuthProvider>
  )
  expect(queryAllByTestId("HabitListItem")).toHaveLength(0)
  expect(queryAllByTestId("HabitListItemMobile")).toHaveLength(0)
})

test("Today: Render with some habit data inside context", () => {
  const { queryAllByTestId } = render(
    <AuthProvider
      value={{
        data: {
          habits: {
            lastUpdate: null,
            checked: data.habits.checked,
            unchecked: data.habits.unchecked,
            map: [],
          },
        },
      }}
    >
      <Today path="/app/" handleAddHabit={() => {}} />
    </AuthProvider>
  )
  expect(queryAllByTestId("HabitListItem")).toHaveLength(9)
  expect(queryAllByTestId("HabitListItemMobile")).toHaveLength(9)
})
