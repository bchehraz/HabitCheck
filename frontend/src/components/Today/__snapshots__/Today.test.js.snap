// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Today.js Snapshot Test Today: Snapshot test 1`] = `
Array [
  <header>
    <div
      className="sc-bdVaJa hgqvdk"
    >
      <h1>
        Today
      </h1>
      <p>
        Saturday, July 4 2020
      </p>
    </div>
  </header>,
  <div
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexFlow": "column nowrap",
        "margin": "0 auto 5rem",
        "maxWidth": "100%",
        "padding": 0,
        "position": "relative",
      }
    }
  >
    <main>
      <div
        className="sc-EHOje jIKgRi habitList"
      >
        <div>
          <div
            className="sc-iwsKbI jbDXSk addHabitButton"
            onClick={[Function]}
          >
            <form
              className="sc-gqjmRU gUSZrW form"
            >
              <h4
                className="sc-jzJRlG bNsmGJ"
              >
                New Habit
              </h4>
              <div
                className="sc-VigVT ceejnY"
              >
                <input
                  className="sc-jTzLTM fTfcBt habitTitleInput"
                  onChange={[Function]}
                  type="text"
                  value=""
                />
                <button
                  className="sc-fjdhpX hBpzPe"
                  onClick={[Function]}
                >
                  <svg
                    className="buttonIcon"
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    strokeWidth="0"
                    style={
                      Object {
                        "color": undefined,
                      }
                    }
                    viewBox="0 0 16 16"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clipRule="evenodd"
                      d="M13.854 3.646a.5.5 0 010 .708l-7 7a.5.5 0 01-.708 0l-3.5-3.5a.5.5 0 11.708-.708L6.5 10.293l6.646-6.647a.5.5 0 01.708 0z"
                      fillRule="evenodd"
                    />
                  </svg>
                </button>
              </div>
            </form>
            <div
              className="sc-gZMcBi glpbXt iconContainerOpen"
              data-testid="AddHabitButtonClose"
            >
              <svg
                className="icon"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                viewBox="0 0 448 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
                />
              </svg>
            </div>
          </div>
          <div
            className="sc-dnqmqq hhhkoQ"
            onClick={[Function]}
          />
        </div>
      </div>
      <div
        className="sc-csuQGl eGzWpx"
      />
      <div
        className="sc-dnqmqq loabua"
        onClick={[Function]}
      />
    </main>
  </div>,
  <div
    className="sc-bwzfXH hCiwAb"
  >
    <div
      className="sc-htpNat hMjIwR footer"
    >
      <a
        href="/app/"
      >
        <div
          className="sc-bxivhb bEdBIT today"
        >
          <div
            className="sc-ifAKCX biWapd"
          />
          <svg
            className="todayIcon"
            fill="none"
            height={40}
            size={40}
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width={40}
            xmlns="http://www.w3.org/2000/svg"
          >
            <polyline
              points="9 11 12 14 22 4"
            />
            <path
              d="M21 12v7a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h11"
            />
          </svg>
        </div>
      </a>
      <a
        href="/app/journal"
      >
        <div
          className="sc-bxivhb dQxvwo yourJournal"
        >
          <svg
            className="journalIcon"
            fill="currentColor"
            height={32}
            size={32}
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 576 512"
            width={32}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M542.22 32.05c-54.8 3.11-163.72 14.43-230.96 55.59-4.64 2.84-7.27 7.89-7.27 13.17v363.87c0 11.55 12.63 18.85 23.28 13.49 69.18-34.82 169.23-44.32 218.7-46.92 16.89-.89 30.02-14.43 30.02-30.66V62.75c.01-17.71-15.35-31.74-33.77-30.7zM264.73 87.64C197.5 46.48 88.58 35.17 33.78 32.05 15.36 31.01 0 45.04 0 62.75V400.6c0 16.24 13.13 29.78 30.02 30.66 49.49 2.6 149.59 12.11 218.77 46.95 10.62 5.35 23.21-1.94 23.21-13.46V100.63c0-5.29-2.62-10.14-7.27-12.99z"
            />
          </svg>
        </div>
      </a>
      <a
        href="/app/settings"
      >
        <div
          className="sc-bxivhb dQxvwo settings"
        >
          <svg
            fill="none"
            height={32}
            size={32}
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width={32}
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12"
              cy="12"
              r="3"
            />
            <path
              d="M19.4 15a1.65 1.65 0 0 0 .33 1.82l.06.06a2 2 0 0 1 0 2.83 2 2 0 0 1-2.83 0l-.06-.06a1.65 1.65 0 0 0-1.82-.33 1.65 1.65 0 0 0-1 1.51V21a2 2 0 0 1-2 2 2 2 0 0 1-2-2v-.09A1.65 1.65 0 0 0 9 19.4a1.65 1.65 0 0 0-1.82.33l-.06.06a2 2 0 0 1-2.83 0 2 2 0 0 1 0-2.83l.06-.06a1.65 1.65 0 0 0 .33-1.82 1.65 1.65 0 0 0-1.51-1H3a2 2 0 0 1-2-2 2 2 0 0 1 2-2h.09A1.65 1.65 0 0 0 4.6 9a1.65 1.65 0 0 0-.33-1.82l-.06-.06a2 2 0 0 1 0-2.83 2 2 0 0 1 2.83 0l.06.06a1.65 1.65 0 0 0 1.82.33H9a1.65 1.65 0 0 0 1-1.51V3a2 2 0 0 1 2-2 2 2 0 0 1 2 2v.09a1.65 1.65 0 0 0 1 1.51 1.65 1.65 0 0 0 1.82-.33l.06-.06a2 2 0 0 1 2.83 0 2 2 0 0 1 0 2.83l-.06.06a1.65 1.65 0 0 0-.33 1.82V9a1.65 1.65 0 0 0 1.51 1H21a2 2 0 0 1 2 2 2 2 0 0 1-2 2h-.09a1.65 1.65 0 0 0-1.51 1z"
            />
          </svg>
        </div>
      </a>
    </div>
  </div>,
]
`;
